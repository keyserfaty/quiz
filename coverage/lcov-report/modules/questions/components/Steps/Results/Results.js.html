<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for modules/questions/components/Steps/Results/Results.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">modules/questions/components/Steps/Results</a> Results.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import { View, Text, ScrollView } from 'react-native'
&nbsp;
import StepTemplate from '../../StepTemplate'
import Button from '../../../../../components/Button'
&nbsp;
import styles from './statics/styles'
&nbsp;
const Results = props =&gt; {
  const { onRestart, responses, correctResponses } = props
&nbsp;
  const header = (
    &lt;View style={styles.header}&gt;
      &lt;Text style={styles.title}&gt;You scored&lt;/Text&gt;
      &lt;View style={styles.counter}&gt;
        &lt;Text&gt;{correctResponses} / 10&lt;/Text&gt;
      &lt;/View&gt;
    &lt;/View&gt;
  )
&nbsp;
  const footer = (
    &lt;Button primary onPress={onRestart}&gt;Play again?&lt;/Button&gt;
  )
&nbsp;
  return (
    &lt;StepTemplate header={header} footer={footer}&gt;
      &lt;ScrollView&gt;
        &lt;View&gt;
          { responses.map((answer, i) =&gt; (
            &lt;View key={i} style={styles.response}&gt;
              &lt;Text style={styles.answer}&gt;{answer.correct ? '✅' : <span class="branch-1 cbranch-no" title="branch not covered" >'❌'}</span>&lt;/Text&gt;
              &lt;Text&gt;{answer.question.replace(/&amp;quot;/g, '"')}&lt;/Text&gt;
            &lt;/View&gt;
          ))}
        &lt;/View&gt;
      &lt;/ScrollView&gt;
    &lt;/StepTemplate&gt;
  )
}
&nbsp;
export default Results</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Nov 11 2018 06:29:52 GMT-0500 (EST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
