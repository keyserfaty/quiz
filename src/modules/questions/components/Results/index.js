import React from 'react'
import { View, Text } from 'react-native'

import Button from '../../../../components/Button'

import styles from './statics/styles'

const Results = props => {
  const { onNextStep, responses, score } = props

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Text>You scored</Text>
        <Text>{score} / 10</Text>
      </View>
      <View style={styles.body}>
        { responses.map((answer, i) => (
          <View key={i} style={styles.response}>
            <Text>{answer.correct ? '+' : '-'}</Text>
            <Text>{answer.question}</Text>
          </View>
        )) }
      </View>
      <View style={styles.footer}>
        <Button primary onPress={() => onNextStep(1)}>Play again?</Button>
      </View>
    </View>
  )
}

export default Results