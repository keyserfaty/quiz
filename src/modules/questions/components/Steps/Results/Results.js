import React from 'react'
import PropTypes from 'prop-types'
import { View, Text, ScrollView } from 'react-native'

import StepTemplate from '../../StepTemplate'
import Button from '../../../../../components/Button'

import styles from './statics/styles'

const Results = props => {
  const { onRestart, responses, correctResponses } = props

  const header = (
    <View style={styles.header}>
      <Text style={styles.title}>You scored</Text>
      <View style={styles.counter}>
        <Text>{correctResponses} / 10</Text>
      </View>
    </View>
  )

  const footer = (
    <Button primary onPress={onRestart}>Play again?</Button>
  )

  return (
    <StepTemplate header={header} footer={footer}>
      <ScrollView>
        <View>
          { responses.map((answer, i) => (
            <View key={i} style={styles.response}>
              <Text style={styles.answer}>{answer.correct ? '✅' : '❌'}</Text>
              <Text>{answer.question.replace(/&quot;/g, '"')}</Text>
            </View>
          ))}
        </View>
      </ScrollView>
    </StepTemplate>
  )
}

Results.propTypes = {
  onRestart: PropTypes.func,
  responses: PropTypes.array,
  correctResponses: PropTypes.number
}

export default Results