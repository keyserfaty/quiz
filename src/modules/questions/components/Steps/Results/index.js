import React from 'react'
import { View, Text } from 'react-native'

import StepTemplate from '../../StepTemplate'
import Button from '../../../../../components/Button'

import styles from './statics/styles'

const Results = props => {
  const { onNextStep, responses, score } = props

  const header = (
    <View style={styles.header}>
      <Text>You scored</Text>
      <Text>{score} / 10</Text>
    </View>
  )

  // TODO: should refresh questions
  const footer = (
    <Button primary onPress={() => onNextStep(0)}>Play again?</Button>
  )

  return (
    <StepTemplate header={header} footer={footer}>
      { responses.map((answer, i) => (
        <View key={i} style={styles.response}>
          <Text>{answer.correct ? '+' : '-'}</Text>
          <Text>{answer.question}</Text>
        </View>
      )) }
    </StepTemplate>
  )
}

export default Results