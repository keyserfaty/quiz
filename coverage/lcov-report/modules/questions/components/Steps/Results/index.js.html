<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for modules/questions/components/Steps/Results/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">modules/questions/components/Steps/Results</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">16.67% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">16.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/6</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
&nbsp;
import Results from './Results'
&nbsp;
class ResultsContainer extends React.Component {
  onRestart = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >    if (this.props.onNextStep &amp;&amp; this.props.onfetchQuestions) {</span>
<span class="cstat-no" title="statement not covered" >      this.props.onfetchQuestions()</span>
<span class="cstat-no" title="statement not covered" >      this.props.onNextStep(0)</span>
    }
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  co</span>mponentWillUnmount () {
<span class="cstat-no" title="statement not covered" >    if (this.props.onCleanResponses) {</span>
<span class="cstat-no" title="statement not covered" >      this.props.onCleanResponses()</span>
    }
  }
&nbsp;
  render () {
    return (
      &lt;Results
        {...this.props}
        onRestart={this.onRestart}
      /&gt;
    )
  }
}
&nbsp;
export default ResultsContainer</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Nov 11 2018 06:19:18 GMT-0500 (EST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
