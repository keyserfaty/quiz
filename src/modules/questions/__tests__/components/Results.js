import React from 'react'
import Results from '../../components/Results/index'

import renderer from 'react-test-renderer'

test('Should return a List element', () => {
  const props = {
    responses: [
      {
        correct: true,
        question: 'Some question 1'
      },
      {
        correct: false,
        question: 'Some question 2'
      }
    ],
    score: 3
  }
  
  const tree = renderer.create(<Results {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})