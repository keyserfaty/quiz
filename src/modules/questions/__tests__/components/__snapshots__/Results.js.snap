// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should return a List element 1`] = `
<View
  style={Object {}}
>
  <View
    style={Object {}}
  >
    <Text>
      You scored
    </Text>
    <Text>
      3
       / 10
    </Text>
  </View>
  <View
    style={Object {}}
  >
    <View
      style={Object {}}
    >
      <Text>
        +
      </Text>
      <Text>
        Some question 1
      </Text>
    </View>
    <View
      style={Object {}}
    >
      <Text>
        -
      </Text>
      <Text>
        Some question 2
      </Text>
    </View>
  </View>
  <View
    style={Object {}}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderRadius": 50,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 30,
          "paddingVertical": 15,
          "width": 235,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Poppins-Regular",
              "fontSize": 18,
              "letterSpacing": 2,
              "textAlign": "center",
            },
            Object {
              "color": "rgba(55,175,255,0.87)",
            },
            undefined,
          ]
        }
      >
        PLAY AGAIN?
      </Text>
    </View>
  </View>
</View>
`;
